site/src/Consumer/EmailNotificationConsumer.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace App\Consumer;
+
+use App\Util\Notification;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Class EmailNotificationConsumer
+ *
+ * This class is responsible for sending out Notifications
+ * to the user's email
+ *
+ * @author Aquib Baig <aquibbaig97@gmail.com>
+ */
+class EmailNotificationConsumer extends AbstractNotificationConsumer
+{
+    /**
+     * EmailNotificationConsumer constructor.
+     * @param LoggerInterface $logger
+     */
+    public function __construct(LoggerInterface $logger)
+    {
+        parent::__construct($logger);
+    }
+
+    /**
+     * Should the Notification be handled by this Consumer?
+     *
+     * @param Notification $notification
+     *
+     * @return bool
+     */
+    protected function shouldHandle(Notification $notification): bool
+    {
+        // TODO: Check if this Consumer should actually handle the Notification
+        return true;
+    }
```

Comment:
You can remove this function if you implement it in the abstract class.

Reply:
@imphil Please correct me out here. How do we implement logic that makes a particular Consumer handle a specific type of notification if we implement the `shouldHandle()` in the abstract class? Will we override that later in the consumers separately when we implement it?

Reply:
@imphil Otherwise I have pushed the requested changes

Reply:
you override it if it should do something else than the default (which is implemented in the abstract class)